### PR TITLE
Fix: Make sure that expect matches the runtime type when instrumented

### PR DESCRIPTION
**Pull request by kasperpeulen**
Closes https://github.com/storybookjs/jest/issues/32

## Change Type

Indicate the type of change your pull request is:

- [ ] `maintenance`
- [ ] `documentation`
- [x] `patch`
- [ ] `minor`
- [ ] `major`

This is a fix, but also in some sense a breaking change... 

<details>
  <summary>📦 Published PR as canary version: <code>0.2.0--canary.34.b637a39.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/jest@0.2.0--canary.34.b637a39.0
  # or 
  yarn add @storybook/jest@0.2.0--canary.34.b637a39.0
  ```
</details>

### CODE REVIEW

**Review by yannbf — APPROVED** [review]
This is amazing! ❤️

File: package.json [hunk 1]
```diff
@@ -34,6 +35,7 @@
     "auto": "^10.37.6",
     "util": "^0.12.5",
     "expect": "^27.3.1",
+    "prettier": "^2.8.8",
```
Comment (IanVS) [inline package.json:38]: This seems unrelated to the other changes here, was it intentional?

Comment (kasperpeulen) [reply package.json:38]: Ah, yeah, I couldn't format my changes! I guess it is okay, to add it.
